middleware/csrf/session_manager.go

```diff
@@ -44,25 +55,39 @@ func (m *sessionManager) getRaw(c fiber.Ctx, key string, raw []byte) []byte {
 
 // set token in session
 func (m *sessionManager) setRaw(c fiber.Ctx, key string, raw []byte, exp time.Duration) {
-	sess, err := m.session.Get(c)
-	if err != nil {
-		return
-	}
-	// the key is crucial in crsf and sometimes a reference to another value which can be reused later(pool/unsafe values concept), so a copy is made here
-	sess.Set(m.key, &Token{key, raw, time.Now().Add(exp)})
-	if err := sess.Save(); err != nil {
-		log.Warn("csrf: failed to save session: ", err)
+	sess := session.FromContext(c)
+	if sess != nil {
+		// the key is crucial in crsf and sometimes a reference to another value which can be reused later(pool/unsafe values concept), so a copy is made here
+		sess.Set(m.key, &Token{key, raw, time.Now().Add(exp)})
+	} else {
+		// Try to get the session from the store
+		storeSess, err := m.session.Get(c)
+		if err != nil {
+			// Handle error
+			return
+		}
+		storeSess.Set(m.key, &Token{key, raw, time.Now().Add(exp)})
+		if err := storeSess.Save(); err != nil {
+			log.Warn("csrf: failed to save session: ", err)
+		}
 	}
 }
 
 // delete token from session
 func (m *sessionManager) delRaw(c fiber.Ctx) {
-	sess, err := m.session.Get(c)
-	if err != nil {
-		return
-	}
-	sess.Delete(m.key)
-	if err := sess.Save(); err != nil {
-		log.Warn("csrf: failed to save session: ", err)
+	sess := session.FromContext(c)
+	if sess != nil {
+		sess.Delete(m.key)
+	} else {
+		// Try to get the session from the store
+		storeSess, err := m.session.Get(c)
+		if err != nil {
```

Comment:
😢

Reply:
Yep... Unfortunalty `middleware/csrf/storage_manager.go` also ignored errors in the same way eg. `_ = m.storage.Set(key, raw, exp) //nolint:errcheck // TODO: Do not ignore error`

Reply:
let's go to solve together

Reply:
The "safe" option appears to be failing the CSRF, so it's likely the correct result already. We could incorporate logging to improve things.